// @hathora/hathora-cloud-sdk/dist/esm/models/AppConfig.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { ApplicationAuthConfigurationFromJSON, ApplicationAuthConfigurationToJSON, } from './ApplicationAuthConfiguration';
/**
 * Check if a given object implements the AppConfig interface.
 */
export function instanceOfAppConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "authConfiguration" in value;
    isInstance = isInstance && "appName" in value;
    return isInstance;
}
export function AppConfigFromJSON(json) {
    return AppConfigFromJSONTyped(json, false);
}
export function AppConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'authConfiguration': ApplicationAuthConfigurationFromJSON(json['authConfiguration']),
        'appName': json['appName'],
    };
}
export function AppConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'authConfiguration': ApplicationAuthConfigurationToJSON(value.authConfiguration),
        'appName': value.appName,
    };
}
