// @hathora/hathora-cloud-sdk/dist/esm/models/ApplicationAuthConfigurationGoogle.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the ApplicationAuthConfigurationGoogle interface.
 */
export function instanceOfApplicationAuthConfigurationGoogle(value) {
    let isInstance = true;
    isInstance = isInstance && "clientId" in value;
    return isInstance;
}
export function ApplicationAuthConfigurationGoogleFromJSON(json) {
    return ApplicationAuthConfigurationGoogleFromJSONTyped(json, false);
}
export function ApplicationAuthConfigurationGoogleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'clientId': json['clientId'],
    };
}
export function ApplicationAuthConfigurationGoogleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'clientId': value.clientId,
    };
}
