// @hathora/hathora-cloud-sdk/dist/esm/models/ApplicationAuthConfiguration.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { ApplicationAuthConfigurationGoogleFromJSON, ApplicationAuthConfigurationGoogleToJSON, } from './ApplicationAuthConfigurationGoogle';
/**
 * Check if a given object implements the ApplicationAuthConfiguration interface.
 */
export function instanceOfApplicationAuthConfiguration(value) {
    let isInstance = true;
    return isInstance;
}
export function ApplicationAuthConfigurationFromJSON(json) {
    return ApplicationAuthConfigurationFromJSONTyped(json, false);
}
export function ApplicationAuthConfigurationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !exists(json, 'google') ? undefined : ApplicationAuthConfigurationGoogleFromJSON(json['google']),
        'nickname': !exists(json, 'nickname') ? undefined : json['nickname'],
        'anonymous': !exists(json, 'anonymous') ? undefined : json['anonymous'],
    };
}
export function ApplicationAuthConfigurationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': ApplicationAuthConfigurationGoogleToJSON(value.google),
        'nickname': value.nickname,
        'anonymous': value.anonymous,
    };
}
