// @hathora/hathora-cloud-sdk/dist/esm/models/Build.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { BuildRegionalContainerTagsInnerFromJSON, BuildRegionalContainerTagsInnerToJSON, } from './BuildRegionalContainerTagsInner';
/**
 * @export
 */
export const BuildStatusEnum = {
    Created: 'created',
    Running: 'running',
    Succeeded: 'succeeded',
    Failed: 'failed'
};
/**
 * Check if a given object implements the Build interface.
 */
export function instanceOfBuild(value) {
    let isInstance = true;
    isInstance = isInstance && "regionalContainerTags" in value;
    isInstance = isInstance && "imageSize" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "deletedAt" in value;
    isInstance = isInstance && "finishedAt" in value;
    isInstance = isInstance && "startedAt" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "buildId" in value;
    isInstance = isInstance && "appId" in value;
    return isInstance;
}
export function BuildFromJSON(json) {
    return BuildFromJSONTyped(json, false);
}
export function BuildFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'regionalContainerTags': (json['regionalContainerTags'].map(BuildRegionalContainerTagsInnerFromJSON)),
        'imageSize': json['imageSize'],
        'status': json['status'],
        'deletedAt': (json['deletedAt'] === null ? null : new Date(json['deletedAt'])),
        'finishedAt': (json['finishedAt'] === null ? null : new Date(json['finishedAt'])),
        'startedAt': (json['startedAt'] === null ? null : new Date(json['startedAt'])),
        'createdAt': (new Date(json['createdAt'])),
        'createdBy': json['createdBy'],
        'buildId': json['buildId'],
        'appId': json['appId'],
    };
}
export function BuildToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'regionalContainerTags': (value.regionalContainerTags.map(BuildRegionalContainerTagsInnerToJSON)),
        'imageSize': value.imageSize,
        'status': value.status,
        'deletedAt': (value.deletedAt === null ? null : value.deletedAt.toISOString()),
        'finishedAt': (value.finishedAt === null ? null : value.finishedAt.toISOString()),
        'startedAt': (value.startedAt === null ? null : value.startedAt.toISOString()),
        'createdAt': (value.createdAt.toISOString()),
        'createdBy': value.createdBy,
        'buildId': value.buildId,
        'appId': value.appId,
    };
}
