// @hathora/hathora-cloud-sdk/dist/esm/models/ConnectionInfo.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { TransportTypeFromJSON, TransportTypeToJSON, } from './TransportType';
/**
 * @export
 */
export const ConnectionInfoStatusEnum = {
    Active: 'active'
};
/**
 * Check if a given object implements the ConnectionInfo interface.
 */
export function instanceOfConnectionInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "roomId" in value;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "host" in value;
    return isInstance;
}
export function ConnectionInfoFromJSON(json) {
    return ConnectionInfoFromJSONTyped(json, false);
}
export function ConnectionInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'status': json['status'],
        'roomId': json['roomId'],
        'transportType': TransportTypeFromJSON(json['transportType']),
        'port': json['port'],
        'host': json['host'],
    };
}
export function ConnectionInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'status': value.status,
        'roomId': value.roomId,
        'transportType': TransportTypeToJSON(value.transportType),
        'port': value.port,
        'host': value.host,
    };
}
