// @hathora/hathora-cloud-sdk/dist/esm/models/CreateAppRequestAuthConfigurationGoogle.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the CreateAppRequestAuthConfigurationGoogle interface.
 */
export function instanceOfCreateAppRequestAuthConfigurationGoogle(value) {
    let isInstance = true;
    isInstance = isInstance && "clientId" in value;
    return isInstance;
}
export function CreateAppRequestAuthConfigurationGoogleFromJSON(json) {
    return CreateAppRequestAuthConfigurationGoogleFromJSONTyped(json, false);
}
export function CreateAppRequestAuthConfigurationGoogleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'clientId': json['clientId'],
    };
}
export function CreateAppRequestAuthConfigurationGoogleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'clientId': value.clientId,
    };
}
