// @hathora/hathora-cloud-sdk/dist/esm/models/CreateAppRequestAuthConfiguration.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { CreateAppRequestAuthConfigurationGoogleFromJSON, CreateAppRequestAuthConfigurationGoogleToJSON, } from './CreateAppRequestAuthConfigurationGoogle';
/**
 * Check if a given object implements the CreateAppRequestAuthConfiguration interface.
 */
export function instanceOfCreateAppRequestAuthConfiguration(value) {
    let isInstance = true;
    return isInstance;
}
export function CreateAppRequestAuthConfigurationFromJSON(json) {
    return CreateAppRequestAuthConfigurationFromJSONTyped(json, false);
}
export function CreateAppRequestAuthConfigurationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !exists(json, 'google') ? undefined : CreateAppRequestAuthConfigurationGoogleFromJSON(json['google']),
        'nickname': !exists(json, 'nickname') ? undefined : json['nickname'],
        'anonymous': !exists(json, 'anonymous') ? undefined : json['anonymous'],
    };
}
export function CreateAppRequestAuthConfigurationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': CreateAppRequestAuthConfigurationGoogleToJSON(value.google),
        'nickname': value.nickname,
        'anonymous': value.anonymous,
    };
}
