// @hathora/hathora-cloud-sdk/dist/esm/models/CreateAppRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { CreateAppRequestAuthConfigurationFromJSON, CreateAppRequestAuthConfigurationToJSON, } from './CreateAppRequestAuthConfiguration';
/**
 * Check if a given object implements the CreateAppRequest interface.
 */
export function instanceOfCreateAppRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "authConfiguration" in value;
    isInstance = isInstance && "appName" in value;
    return isInstance;
}
export function CreateAppRequestFromJSON(json) {
    return CreateAppRequestFromJSONTyped(json, false);
}
export function CreateAppRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'authConfiguration': CreateAppRequestAuthConfigurationFromJSON(json['authConfiguration']),
        'appName': json['appName'],
    };
}
export function CreateAppRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'authConfiguration': CreateAppRequestAuthConfigurationToJSON(value.authConfiguration),
        'appName': value.appName,
    };
}
