// @hathora/hathora-cloud-sdk/dist/esm/models/DeploymentConfigEnvInner.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the DeploymentConfigEnvInner interface.
 */
export function instanceOfDeploymentConfigEnvInner(value) {
    let isInstance = true;
    isInstance = isInstance && "value" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
export function DeploymentConfigEnvInnerFromJSON(json) {
    return DeploymentConfigEnvInnerFromJSONTyped(json, false);
}
export function DeploymentConfigEnvInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': json['value'],
        'name': json['name'],
    };
}
export function DeploymentConfigEnvInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': value.value,
        'name': value.name,
    };
}
