// @hathora/hathora-cloud-sdk/dist/esm/models/DiscoveryResponseInner.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { RegionFromJSON, RegionToJSON, } from './Region';
/**
 * Check if a given object implements the DiscoveryResponseInner interface.
 */
export function instanceOfDiscoveryResponseInner(value) {
    let isInstance = true;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "host" in value;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
export function DiscoveryResponseInnerFromJSON(json) {
    return DiscoveryResponseInnerFromJSONTyped(json, false);
}
export function DiscoveryResponseInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'port': json['port'],
        'host': json['host'],
        'region': RegionFromJSON(json['region']),
    };
}
export function DiscoveryResponseInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'port': value.port,
        'host': value.host,
        'region': RegionToJSON(value.region),
    };
}
