// @hathora/hathora-cloud-sdk/dist/esm/models/Lobby.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { RegionFromJSON, RegionToJSON, } from './Region';
/**
 * @export
 */
export const LobbyVisibilityEnum = {
    Private: 'private',
    Public: 'public',
    Local: 'local'
};
/**
 * Check if a given object implements the Lobby interface.
 */
export function instanceOfLobby(value) {
    let isInstance = true;
    isInstance = isInstance && "initialConfig" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "local" in value;
    isInstance = isInstance && "visibility" in value;
    isInstance = isInstance && "region" in value;
    isInstance = isInstance && "roomId" in value;
    isInstance = isInstance && "appId" in value;
    return isInstance;
}
export function LobbyFromJSON(json) {
    return LobbyFromJSONTyped(json, false);
}
export function LobbyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'state': !exists(json, 'state') ? undefined : json['state'],
        'initialConfig': json['initialConfig'],
        'createdAt': (new Date(json['createdAt'])),
        'createdBy': json['createdBy'],
        'local': json['local'],
        'visibility': json['visibility'],
        'region': RegionFromJSON(json['region']),
        'roomId': json['roomId'],
        'appId': json['appId'],
    };
}
export function LobbyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'state': value.state,
        'initialConfig': value.initialConfig,
        'createdAt': (value.createdAt.toISOString()),
        'createdBy': value.createdBy,
        'local': value.local,
        'visibility': value.visibility,
        'region': RegionToJSON(value.region),
        'roomId': value.roomId,
        'appId': value.appId,
    };
}
