// @hathora/hathora-cloud-sdk/dist/esm/models/LoginNicknameRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the LoginNicknameRequest interface.
 */
export function instanceOfLoginNicknameRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "nickname" in value;
    return isInstance;
}
export function LoginNicknameRequestFromJSON(json) {
    return LoginNicknameRequestFromJSONTyped(json, false);
}
export function LoginNicknameRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'nickname': json['nickname'],
    };
}
export function LoginNicknameRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'nickname': value.nickname,
    };
}
