// @hathora/hathora-cloud-sdk/dist/esm/models/LoginGoogleRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the LoginGoogleRequest interface.
 */
export function instanceOfLoginGoogleRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "idToken" in value;
    return isInstance;
}
export function LoginGoogleRequestFromJSON(json) {
    return LoginGoogleRequestFromJSONTyped(json, false);
}
export function LoginGoogleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'idToken': json['idToken'],
    };
}
export function LoginGoogleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'idToken': value.idToken,
    };
}
