// @hathora/hathora-cloud-sdk/dist/esm/models/LoginAnonymous200Response.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the LoginAnonymous200Response interface.
 */
export function instanceOfLoginAnonymous200Response(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
export function LoginAnonymous200ResponseFromJSON(json) {
    return LoginAnonymous200ResponseFromJSONTyped(json, false);
}
export function LoginAnonymous200ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'token': json['token'],
    };
}
export function LoginAnonymous200ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'token': value.token,
    };
}
