// @hathora/hathora-cloud-sdk/dist/esm/models/PickRoomExcludeKeyofRoomAllocations.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { RoomAllocationFromJSON, RoomAllocationToJSON, } from './RoomAllocation';
import { RoomStatusFromJSON, RoomStatusToJSON, } from './RoomStatus';
/**
 * Check if a given object implements the PickRoomExcludeKeyofRoomAllocations interface.
 */
export function instanceOfPickRoomExcludeKeyofRoomAllocations(value) {
    let isInstance = true;
    isInstance = isInstance && "appId" in value;
    isInstance = isInstance && "roomId" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "currentAllocation" in value;
    return isInstance;
}
export function PickRoomExcludeKeyofRoomAllocationsFromJSON(json) {
    return PickRoomExcludeKeyofRoomAllocationsFromJSONTyped(json, false);
}
export function PickRoomExcludeKeyofRoomAllocationsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'appId': json['appId'],
        'roomId': json['roomId'],
        'status': RoomStatusFromJSON(json['status']),
        'currentAllocation': RoomAllocationFromJSON(json['currentAllocation']),
    };
}
export function PickRoomExcludeKeyofRoomAllocationsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'appId': value.appId,
        'roomId': value.roomId,
        'status': RoomStatusToJSON(value.status),
        'currentAllocation': RoomAllocationToJSON(value.currentAllocation),
    };
}
