// @hathora/hathora-cloud-sdk/dist/esm/models/ProcessWithRoomsAllOf.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { PickRoomExcludeKeyofRoomAllocationsFromJSON, PickRoomExcludeKeyofRoomAllocationsToJSON, } from './PickRoomExcludeKeyofRoomAllocations';
/**
 * Check if a given object implements the ProcessWithRoomsAllOf interface.
 */
export function instanceOfProcessWithRoomsAllOf(value) {
    let isInstance = true;
    isInstance = isInstance && "rooms" in value;
    isInstance = isInstance && "totalRooms" in value;
    return isInstance;
}
export function ProcessWithRoomsAllOfFromJSON(json) {
    return ProcessWithRoomsAllOfFromJSONTyped(json, false);
}
export function ProcessWithRoomsAllOfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'rooms': (json['rooms'].map(PickRoomExcludeKeyofRoomAllocationsFromJSON)),
        'totalRooms': json['totalRooms'],
    };
}
export function ProcessWithRoomsAllOfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'rooms': (value.rooms.map(PickRoomExcludeKeyofRoomAllocationsToJSON)),
        'totalRooms': value.totalRooms,
    };
}
