// @hathora/hathora-cloud-sdk/dist/esm/models/RecordPartialMetricNameMetricValueArray.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime';
import { MetricValueFromJSON, MetricValueToJSON, } from './MetricValue';
/**
 * Check if a given object implements the RecordPartialMetricNameMetricValueArray interface.
 */
export function instanceOfRecordPartialMetricNameMetricValueArray(value) {
    let isInstance = true;
    return isInstance;
}
export function RecordPartialMetricNameMetricValueArrayFromJSON(json) {
    return RecordPartialMetricNameMetricValueArrayFromJSONTyped(json, false);
}
export function RecordPartialMetricNameMetricValueArrayFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'cpu': !exists(json, 'cpu') ? undefined : (json['cpu'].map(MetricValueFromJSON)),
        'memory': !exists(json, 'memory') ? undefined : (json['memory'].map(MetricValueFromJSON)),
        'rateEgress': !exists(json, 'rate_egress') ? undefined : (json['rate_egress'].map(MetricValueFromJSON)),
        'totalEgress': !exists(json, 'total_egress') ? undefined : (json['total_egress'].map(MetricValueFromJSON)),
    };
}
export function RecordPartialMetricNameMetricValueArrayToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'cpu': value.cpu === undefined ? undefined : (value.cpu.map(MetricValueToJSON)),
        'memory': value.memory === undefined ? undefined : (value.memory.map(MetricValueToJSON)),
        'rate_egress': value.rateEgress === undefined ? undefined : (value.rateEgress.map(MetricValueToJSON)),
        'total_egress': value.totalEgress === undefined ? undefined : (value.totalEgress.map(MetricValueToJSON)),
    };
}
