// @hathora/hathora-cloud-sdk/dist/esm/models/SetLobbyStateRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the SetLobbyStateRequest interface.
 */
export function instanceOfSetLobbyStateRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "state" in value;
    return isInstance;
}
export function SetLobbyStateRequestFromJSON(json) {
    return SetLobbyStateRequestFromJSONTyped(json, false);
}
export function SetLobbyStateRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'state': json['state'],
    };
}
export function SetLobbyStateRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'state': value.state,
    };
}
