// @hathora/hathora-cloud-sdk/dist/esm/models/StartingConnectionInfo.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * @export
 */
export const StartingConnectionInfoStatusEnum = {
    Starting: 'starting'
};
/**
 * Check if a given object implements the StartingConnectionInfo interface.
 */
export function instanceOfStartingConnectionInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "roomId" in value;
    return isInstance;
}
export function StartingConnectionInfoFromJSON(json) {
    return StartingConnectionInfoFromJSONTyped(json, false);
}
export function StartingConnectionInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'status': json['status'],
        'roomId': json['roomId'],
    };
}
export function StartingConnectionInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'status': value.status,
        'roomId': value.roomId,
    };
}
