// @hathora/hathora-cloud-sdk/dist/esm/apis/DeploymentV1Api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import * as runtime from '../runtime';
import { DeploymentFromJSON, DeploymentConfigToJSON, } from '../models';
/**
 *
 */
export class DeploymentV1Api extends runtime.BaseAPI {
    /**
     */
    createDeploymentRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters.appId === null || requestParameters.appId === undefined) {
                throw new runtime.RequiredError('appId', 'Required parameter requestParameters.appId was null or undefined when calling createDeployment.');
            }
            if (requestParameters.buildId === null || requestParameters.buildId === undefined) {
                throw new runtime.RequiredError('buildId', 'Required parameter requestParameters.buildId was null or undefined when calling createDeployment.');
            }
            if (requestParameters.deploymentConfig === null || requestParameters.deploymentConfig === undefined) {
                throw new runtime.RequiredError('deploymentConfig', 'Required parameter requestParameters.deploymentConfig was null or undefined when calling createDeployment.');
            }
            const queryParameters = {};
            const headerParameters = {};
            headerParameters['Content-Type'] = 'application/json';
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("auth0", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/deployments/v1/{appId}/create/{buildId}`.replace(`{${"appId"}}`, encodeURIComponent(String(requestParameters.appId))).replace(`{${"buildId"}}`, encodeURIComponent(String(requestParameters.buildId))),
                method: 'POST',
                headers: headerParameters,
                query: queryParameters,
                body: DeploymentConfigToJSON(requestParameters.deploymentConfig),
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => DeploymentFromJSON(jsonValue));
        });
    }
    /**
     */
    createDeployment(appId, buildId, deploymentConfig, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.createDeploymentRaw({ appId: appId, buildId: buildId, deploymentConfig: deploymentConfig }, initOverrides);
            return yield response.value();
        });
    }
    /**
     */
    getDeploymentInfoRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters.appId === null || requestParameters.appId === undefined) {
                throw new runtime.RequiredError('appId', 'Required parameter requestParameters.appId was null or undefined when calling getDeploymentInfo.');
            }
            if (requestParameters.deploymentId === null || requestParameters.deploymentId === undefined) {
                throw new runtime.RequiredError('deploymentId', 'Required parameter requestParameters.deploymentId was null or undefined when calling getDeploymentInfo.');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("auth0", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/deployments/v1/{appId}/info/{deploymentId}`.replace(`{${"appId"}}`, encodeURIComponent(String(requestParameters.appId))).replace(`{${"deploymentId"}}`, encodeURIComponent(String(requestParameters.deploymentId))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => DeploymentFromJSON(jsonValue));
        });
    }
    /**
     */
    getDeploymentInfo(appId, deploymentId, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getDeploymentInfoRaw({ appId: appId, deploymentId: deploymentId }, initOverrides);
            return yield response.value();
        });
    }
    /**
     */
    getDeploymentsRaw(requestParameters, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            if (requestParameters.appId === null || requestParameters.appId === undefined) {
                throw new runtime.RequiredError('appId', 'Required parameter requestParameters.appId was null or undefined when calling getDeployments.');
            }
            const queryParameters = {};
            const headerParameters = {};
            if (this.configuration && this.configuration.accessToken) {
                const token = this.configuration.accessToken;
                const tokenString = yield token("auth0", []);
                if (tokenString) {
                    headerParameters["Authorization"] = `Bearer ${tokenString}`;
                }
            }
            const response = yield this.request({
                path: `/deployments/v1/{appId}/list`.replace(`{${"appId"}}`, encodeURIComponent(String(requestParameters.appId))),
                method: 'GET',
                headers: headerParameters,
                query: queryParameters,
            }, initOverrides);
            return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(DeploymentFromJSON));
        });
    }
    /**
     */
    getDeployments(appId, initOverrides) {
        return __awaiter(this, void 0, void 0, function* () {
            const response = yield this.getDeploymentsRaw({ appId: appId }, initOverrides);
            return yield response.value();
        });
    }
}
