// @hathora/hathora-cloud-sdk/dist/models/ActiveConnectionInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActiveConnectionInfoToJSON = exports.ActiveConnectionInfoFromJSONTyped = exports.ActiveConnectionInfoFromJSON = exports.instanceOfActiveConnectionInfo = exports.ActiveConnectionInfoStatusEnum = void 0;
const TransportType_1 = require("./TransportType");
/**
 * @export
 */
exports.ActiveConnectionInfoStatusEnum = {
    Active: 'active'
};
/**
 * Check if a given object implements the ActiveConnectionInfo interface.
 */
function instanceOfActiveConnectionInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "host" in value;
    isInstance = isInstance && "roomId" in value;
    return isInstance;
}
exports.instanceOfActiveConnectionInfo = instanceOfActiveConnectionInfo;
function ActiveConnectionInfoFromJSON(json) {
    return ActiveConnectionInfoFromJSONTyped(json, false);
}
exports.ActiveConnectionInfoFromJSON = ActiveConnectionInfoFromJSON;
function ActiveConnectionInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'status': json['status'],
        'transportType': (0, TransportType_1.TransportTypeFromJSON)(json['transportType']),
        'port': json['port'],
        'host': json['host'],
        'roomId': json['roomId'],
    };
}
exports.ActiveConnectionInfoFromJSONTyped = ActiveConnectionInfoFromJSONTyped;
function ActiveConnectionInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'status': value.status,
        'transportType': (0, TransportType_1.TransportTypeToJSON)(value.transportType),
        'port': value.port,
        'host': value.host,
        'roomId': value.roomId,
    };
}
exports.ActiveConnectionInfoToJSON = ActiveConnectionInfoToJSON;
