// @hathora/hathora-cloud-sdk/dist/models/AppConfig.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppConfigToJSON = exports.AppConfigFromJSONTyped = exports.AppConfigFromJSON = exports.instanceOfAppConfig = void 0;
const ApplicationAuthConfiguration_1 = require("./ApplicationAuthConfiguration");
/**
 * Check if a given object implements the AppConfig interface.
 */
function instanceOfAppConfig(value) {
    let isInstance = true;
    isInstance = isInstance && "authConfiguration" in value;
    isInstance = isInstance && "appName" in value;
    return isInstance;
}
exports.instanceOfAppConfig = instanceOfAppConfig;
function AppConfigFromJSON(json) {
    return AppConfigFromJSONTyped(json, false);
}
exports.AppConfigFromJSON = AppConfigFromJSON;
function AppConfigFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'authConfiguration': (0, ApplicationAuthConfiguration_1.ApplicationAuthConfigurationFromJSON)(json['authConfiguration']),
        'appName': json['appName'],
    };
}
exports.AppConfigFromJSONTyped = AppConfigFromJSONTyped;
function AppConfigToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'authConfiguration': (0, ApplicationAuthConfiguration_1.ApplicationAuthConfigurationToJSON)(value.authConfiguration),
        'appName': value.appName,
    };
}
exports.AppConfigToJSON = AppConfigToJSON;
