// @hathora/hathora-cloud-sdk/dist/models/ApplicationWithDeployment.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationWithDeploymentToJSON = exports.ApplicationWithDeploymentFromJSONTyped = exports.ApplicationWithDeploymentFromJSON = exports.instanceOfApplicationWithDeployment = void 0;
const ApplicationAuthConfiguration_1 = require("./ApplicationAuthConfiguration");
const Deployment_1 = require("./Deployment");
/**
 * Check if a given object implements the ApplicationWithDeployment interface.
 */
function instanceOfApplicationWithDeployment(value) {
    let isInstance = true;
    isInstance = isInstance && "deletedBy" in value;
    isInstance = isInstance && "deletedAt" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "orgId" in value;
    isInstance = isInstance && "authConfiguration" in value;
    isInstance = isInstance && "appSecret" in value;
    isInstance = isInstance && "appId" in value;
    isInstance = isInstance && "appName" in value;
    isInstance = isInstance && "deployment" in value;
    return isInstance;
}
exports.instanceOfApplicationWithDeployment = instanceOfApplicationWithDeployment;
function ApplicationWithDeploymentFromJSON(json) {
    return ApplicationWithDeploymentFromJSONTyped(json, false);
}
exports.ApplicationWithDeploymentFromJSON = ApplicationWithDeploymentFromJSON;
function ApplicationWithDeploymentFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deletedBy': json['deletedBy'],
        'deletedAt': (json['deletedAt'] === null ? null : new Date(json['deletedAt'])),
        'createdAt': (new Date(json['createdAt'])),
        'createdBy': json['createdBy'],
        'orgId': json['orgId'],
        'authConfiguration': (0, ApplicationAuthConfiguration_1.ApplicationAuthConfigurationFromJSON)(json['authConfiguration']),
        'appSecret': json['appSecret'],
        'appId': json['appId'],
        'appName': json['appName'],
        'deployment': (0, Deployment_1.DeploymentFromJSON)(json['deployment']),
    };
}
exports.ApplicationWithDeploymentFromJSONTyped = ApplicationWithDeploymentFromJSONTyped;
function ApplicationWithDeploymentToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'deletedBy': value.deletedBy,
        'deletedAt': (value.deletedAt === null ? null : value.deletedAt.toISOString()),
        'createdAt': (value.createdAt.toISOString()),
        'createdBy': value.createdBy,
        'orgId': value.orgId,
        'authConfiguration': (0, ApplicationAuthConfiguration_1.ApplicationAuthConfigurationToJSON)(value.authConfiguration),
        'appSecret': value.appSecret,
        'appId': value.appId,
        'appName': value.appName,
        'deployment': (0, Deployment_1.DeploymentToJSON)(value.deployment),
    };
}
exports.ApplicationWithDeploymentToJSON = ApplicationWithDeploymentToJSON;
