// @hathora/hathora-cloud-sdk/dist/models/ApplicationAuthConfiguration.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationAuthConfigurationToJSON = exports.ApplicationAuthConfigurationFromJSONTyped = exports.ApplicationAuthConfigurationFromJSON = exports.instanceOfApplicationAuthConfiguration = void 0;
const runtime_1 = require("../runtime");
const ApplicationAuthConfigurationGoogle_1 = require("./ApplicationAuthConfigurationGoogle");
/**
 * Check if a given object implements the ApplicationAuthConfiguration interface.
 */
function instanceOfApplicationAuthConfiguration(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfApplicationAuthConfiguration = instanceOfApplicationAuthConfiguration;
function ApplicationAuthConfigurationFromJSON(json) {
    return ApplicationAuthConfigurationFromJSONTyped(json, false);
}
exports.ApplicationAuthConfigurationFromJSON = ApplicationAuthConfigurationFromJSON;
function ApplicationAuthConfigurationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !(0, runtime_1.exists)(json, 'google') ? undefined : (0, ApplicationAuthConfigurationGoogle_1.ApplicationAuthConfigurationGoogleFromJSON)(json['google']),
        'nickname': !(0, runtime_1.exists)(json, 'nickname') ? undefined : json['nickname'],
        'anonymous': !(0, runtime_1.exists)(json, 'anonymous') ? undefined : json['anonymous'],
    };
}
exports.ApplicationAuthConfigurationFromJSONTyped = ApplicationAuthConfigurationFromJSONTyped;
function ApplicationAuthConfigurationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': (0, ApplicationAuthConfigurationGoogle_1.ApplicationAuthConfigurationGoogleToJSON)(value.google),
        'nickname': value.nickname,
        'anonymous': value.anonymous,
    };
}
exports.ApplicationAuthConfigurationToJSON = ApplicationAuthConfigurationToJSON;
