// @hathora/hathora-cloud-sdk/dist/models/ApplicationWithDeploymentAllOf.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationWithDeploymentAllOfToJSON = exports.ApplicationWithDeploymentAllOfFromJSONTyped = exports.ApplicationWithDeploymentAllOfFromJSON = exports.instanceOfApplicationWithDeploymentAllOf = void 0;
const Deployment_1 = require("./Deployment");
/**
 * Check if a given object implements the ApplicationWithDeploymentAllOf interface.
 */
function instanceOfApplicationWithDeploymentAllOf(value) {
    let isInstance = true;
    isInstance = isInstance && "deployment" in value;
    return isInstance;
}
exports.instanceOfApplicationWithDeploymentAllOf = instanceOfApplicationWithDeploymentAllOf;
function ApplicationWithDeploymentAllOfFromJSON(json) {
    return ApplicationWithDeploymentAllOfFromJSONTyped(json, false);
}
exports.ApplicationWithDeploymentAllOfFromJSON = ApplicationWithDeploymentAllOfFromJSON;
function ApplicationWithDeploymentAllOfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'deployment': (0, Deployment_1.DeploymentFromJSON)(json['deployment']),
    };
}
exports.ApplicationWithDeploymentAllOfFromJSONTyped = ApplicationWithDeploymentAllOfFromJSONTyped;
function ApplicationWithDeploymentAllOfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'deployment': (0, Deployment_1.DeploymentToJSON)(value.deployment),
    };
}
exports.ApplicationWithDeploymentAllOfToJSON = ApplicationWithDeploymentAllOfToJSON;
