// @hathora/hathora-cloud-sdk/dist/models/ApplicationAuthConfigurationGoogle.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationAuthConfigurationGoogleToJSON = exports.ApplicationAuthConfigurationGoogleFromJSONTyped = exports.ApplicationAuthConfigurationGoogleFromJSON = exports.instanceOfApplicationAuthConfigurationGoogle = void 0;
/**
 * Check if a given object implements the ApplicationAuthConfigurationGoogle interface.
 */
function instanceOfApplicationAuthConfigurationGoogle(value) {
    let isInstance = true;
    isInstance = isInstance && "clientId" in value;
    return isInstance;
}
exports.instanceOfApplicationAuthConfigurationGoogle = instanceOfApplicationAuthConfigurationGoogle;
function ApplicationAuthConfigurationGoogleFromJSON(json) {
    return ApplicationAuthConfigurationGoogleFromJSONTyped(json, false);
}
exports.ApplicationAuthConfigurationGoogleFromJSON = ApplicationAuthConfigurationGoogleFromJSON;
function ApplicationAuthConfigurationGoogleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'clientId': json['clientId'],
    };
}
exports.ApplicationAuthConfigurationGoogleFromJSONTyped = ApplicationAuthConfigurationGoogleFromJSONTyped;
function ApplicationAuthConfigurationGoogleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'clientId': value.clientId,
    };
}
exports.ApplicationAuthConfigurationGoogleToJSON = ApplicationAuthConfigurationGoogleToJSON;
