// @hathora/hathora-cloud-sdk/dist/models/BuildRegionalContainerTagsInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuildRegionalContainerTagsInnerToJSON = exports.BuildRegionalContainerTagsInnerFromJSONTyped = exports.BuildRegionalContainerTagsInnerFromJSON = exports.instanceOfBuildRegionalContainerTagsInner = void 0;
const Region_1 = require("./Region");
/**
 * Check if a given object implements the BuildRegionalContainerTagsInner interface.
 */
function instanceOfBuildRegionalContainerTagsInner(value) {
    let isInstance = true;
    isInstance = isInstance && "containerTag" in value;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
exports.instanceOfBuildRegionalContainerTagsInner = instanceOfBuildRegionalContainerTagsInner;
function BuildRegionalContainerTagsInnerFromJSON(json) {
    return BuildRegionalContainerTagsInnerFromJSONTyped(json, false);
}
exports.BuildRegionalContainerTagsInnerFromJSON = BuildRegionalContainerTagsInnerFromJSON;
function BuildRegionalContainerTagsInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'containerTag': json['containerTag'],
        'region': (0, Region_1.RegionFromJSON)(json['region']),
    };
}
exports.BuildRegionalContainerTagsInnerFromJSONTyped = BuildRegionalContainerTagsInnerFromJSONTyped;
function BuildRegionalContainerTagsInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'containerTag': value.containerTag,
        'region': (0, Region_1.RegionToJSON)(value.region),
    };
}
exports.BuildRegionalContainerTagsInnerToJSON = BuildRegionalContainerTagsInnerToJSON;
