// @hathora/hathora-cloud-sdk/dist/models/ConnectionInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionInfoToJSON = exports.ConnectionInfoFromJSONTyped = exports.ConnectionInfoFromJSON = exports.instanceOfConnectionInfo = exports.ConnectionInfoStatusEnum = void 0;
const TransportType_1 = require("./TransportType");
/**
 * @export
 */
exports.ConnectionInfoStatusEnum = {
    Active: 'active'
};
/**
 * Check if a given object implements the ConnectionInfo interface.
 */
function instanceOfConnectionInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "roomId" in value;
    isInstance = isInstance && "transportType" in value;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "host" in value;
    return isInstance;
}
exports.instanceOfConnectionInfo = instanceOfConnectionInfo;
function ConnectionInfoFromJSON(json) {
    return ConnectionInfoFromJSONTyped(json, false);
}
exports.ConnectionInfoFromJSON = ConnectionInfoFromJSON;
function ConnectionInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'status': json['status'],
        'roomId': json['roomId'],
        'transportType': (0, TransportType_1.TransportTypeFromJSON)(json['transportType']),
        'port': json['port'],
        'host': json['host'],
    };
}
exports.ConnectionInfoFromJSONTyped = ConnectionInfoFromJSONTyped;
function ConnectionInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'status': value.status,
        'roomId': value.roomId,
        'transportType': (0, TransportType_1.TransportTypeToJSON)(value.transportType),
        'port': value.port,
        'host': value.host,
    };
}
exports.ConnectionInfoToJSON = ConnectionInfoToJSON;
