// @hathora/hathora-cloud-sdk/dist/models/CreatePrivateLobbyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePrivateLobbyRequestToJSON = exports.CreatePrivateLobbyRequestFromJSONTyped = exports.CreatePrivateLobbyRequestFromJSON = exports.instanceOfCreatePrivateLobbyRequest = void 0;
const Region_1 = require("./Region");
/**
 * Check if a given object implements the CreatePrivateLobbyRequest interface.
 */
function instanceOfCreatePrivateLobbyRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "initialConfig" in value;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
exports.instanceOfCreatePrivateLobbyRequest = instanceOfCreatePrivateLobbyRequest;
function CreatePrivateLobbyRequestFromJSON(json) {
    return CreatePrivateLobbyRequestFromJSONTyped(json, false);
}
exports.CreatePrivateLobbyRequestFromJSON = CreatePrivateLobbyRequestFromJSON;
function CreatePrivateLobbyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'initialConfig': json['initialConfig'],
        'region': (0, Region_1.RegionFromJSON)(json['region']),
    };
}
exports.CreatePrivateLobbyRequestFromJSONTyped = CreatePrivateLobbyRequestFromJSONTyped;
function CreatePrivateLobbyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'initialConfig': value.initialConfig,
        'region': (0, Region_1.RegionToJSON)(value.region),
    };
}
exports.CreatePrivateLobbyRequestToJSON = CreatePrivateLobbyRequestToJSON;
