// @hathora/hathora-cloud-sdk/dist/models/CreateRoomRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRoomRequestToJSON = exports.CreateRoomRequestFromJSONTyped = exports.CreateRoomRequestFromJSON = exports.instanceOfCreateRoomRequest = void 0;
const Region_1 = require("./Region");
/**
 * Check if a given object implements the CreateRoomRequest interface.
 */
function instanceOfCreateRoomRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
exports.instanceOfCreateRoomRequest = instanceOfCreateRoomRequest;
function CreateRoomRequestFromJSON(json) {
    return CreateRoomRequestFromJSONTyped(json, false);
}
exports.CreateRoomRequestFromJSON = CreateRoomRequestFromJSON;
function CreateRoomRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'region': (0, Region_1.RegionFromJSON)(json['region']),
    };
}
exports.CreateRoomRequestFromJSONTyped = CreateRoomRequestFromJSONTyped;
function CreateRoomRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'region': (0, Region_1.RegionToJSON)(value.region),
    };
}
exports.CreateRoomRequestToJSON = CreateRoomRequestToJSON;
