// @hathora/hathora-cloud-sdk/dist/models/CreateAppRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAppRequestToJSON = exports.CreateAppRequestFromJSONTyped = exports.CreateAppRequestFromJSON = exports.instanceOfCreateAppRequest = void 0;
const CreateAppRequestAuthConfiguration_1 = require("./CreateAppRequestAuthConfiguration");
/**
 * Check if a given object implements the CreateAppRequest interface.
 */
function instanceOfCreateAppRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "authConfiguration" in value;
    isInstance = isInstance && "appName" in value;
    return isInstance;
}
exports.instanceOfCreateAppRequest = instanceOfCreateAppRequest;
function CreateAppRequestFromJSON(json) {
    return CreateAppRequestFromJSONTyped(json, false);
}
exports.CreateAppRequestFromJSON = CreateAppRequestFromJSON;
function CreateAppRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'authConfiguration': (0, CreateAppRequestAuthConfiguration_1.CreateAppRequestAuthConfigurationFromJSON)(json['authConfiguration']),
        'appName': json['appName'],
    };
}
exports.CreateAppRequestFromJSONTyped = CreateAppRequestFromJSONTyped;
function CreateAppRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'authConfiguration': (0, CreateAppRequestAuthConfiguration_1.CreateAppRequestAuthConfigurationToJSON)(value.authConfiguration),
        'appName': value.appName,
    };
}
exports.CreateAppRequestToJSON = CreateAppRequestToJSON;
