// @hathora/hathora-cloud-sdk/dist/models/CreateAppRequestAuthConfigurationGoogle.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAppRequestAuthConfigurationGoogleToJSON = exports.CreateAppRequestAuthConfigurationGoogleFromJSONTyped = exports.CreateAppRequestAuthConfigurationGoogleFromJSON = exports.instanceOfCreateAppRequestAuthConfigurationGoogle = void 0;
/**
 * Check if a given object implements the CreateAppRequestAuthConfigurationGoogle interface.
 */
function instanceOfCreateAppRequestAuthConfigurationGoogle(value) {
    let isInstance = true;
    isInstance = isInstance && "clientId" in value;
    return isInstance;
}
exports.instanceOfCreateAppRequestAuthConfigurationGoogle = instanceOfCreateAppRequestAuthConfigurationGoogle;
function CreateAppRequestAuthConfigurationGoogleFromJSON(json) {
    return CreateAppRequestAuthConfigurationGoogleFromJSONTyped(json, false);
}
exports.CreateAppRequestAuthConfigurationGoogleFromJSON = CreateAppRequestAuthConfigurationGoogleFromJSON;
function CreateAppRequestAuthConfigurationGoogleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'clientId': json['clientId'],
    };
}
exports.CreateAppRequestAuthConfigurationGoogleFromJSONTyped = CreateAppRequestAuthConfigurationGoogleFromJSONTyped;
function CreateAppRequestAuthConfigurationGoogleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'clientId': value.clientId,
    };
}
exports.CreateAppRequestAuthConfigurationGoogleToJSON = CreateAppRequestAuthConfigurationGoogleToJSON;
