// @hathora/hathora-cloud-sdk/dist/models/CreateAppRequestAuthConfiguration.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAppRequestAuthConfigurationToJSON = exports.CreateAppRequestAuthConfigurationFromJSONTyped = exports.CreateAppRequestAuthConfigurationFromJSON = exports.instanceOfCreateAppRequestAuthConfiguration = void 0;
const runtime_1 = require("../runtime");
const CreateAppRequestAuthConfigurationGoogle_1 = require("./CreateAppRequestAuthConfigurationGoogle");
/**
 * Check if a given object implements the CreateAppRequestAuthConfiguration interface.
 */
function instanceOfCreateAppRequestAuthConfiguration(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfCreateAppRequestAuthConfiguration = instanceOfCreateAppRequestAuthConfiguration;
function CreateAppRequestAuthConfigurationFromJSON(json) {
    return CreateAppRequestAuthConfigurationFromJSONTyped(json, false);
}
exports.CreateAppRequestAuthConfigurationFromJSON = CreateAppRequestAuthConfigurationFromJSON;
function CreateAppRequestAuthConfigurationFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'google': !(0, runtime_1.exists)(json, 'google') ? undefined : (0, CreateAppRequestAuthConfigurationGoogle_1.CreateAppRequestAuthConfigurationGoogleFromJSON)(json['google']),
        'nickname': !(0, runtime_1.exists)(json, 'nickname') ? undefined : json['nickname'],
        'anonymous': !(0, runtime_1.exists)(json, 'anonymous') ? undefined : json['anonymous'],
    };
}
exports.CreateAppRequestAuthConfigurationFromJSONTyped = CreateAppRequestAuthConfigurationFromJSONTyped;
function CreateAppRequestAuthConfigurationToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'google': (0, CreateAppRequestAuthConfigurationGoogle_1.CreateAppRequestAuthConfigurationGoogleToJSON)(value.google),
        'nickname': value.nickname,
        'anonymous': value.anonymous,
    };
}
exports.CreateAppRequestAuthConfigurationToJSON = CreateAppRequestAuthConfigurationToJSON;
