// @hathora/hathora-cloud-sdk/dist/models/DeploymentConfigEnvInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentConfigEnvInnerToJSON = exports.DeploymentConfigEnvInnerFromJSONTyped = exports.DeploymentConfigEnvInnerFromJSON = exports.instanceOfDeploymentConfigEnvInner = void 0;
/**
 * Check if a given object implements the DeploymentConfigEnvInner interface.
 */
function instanceOfDeploymentConfigEnvInner(value) {
    let isInstance = true;
    isInstance = isInstance && "value" in value;
    isInstance = isInstance && "name" in value;
    return isInstance;
}
exports.instanceOfDeploymentConfigEnvInner = instanceOfDeploymentConfigEnvInner;
function DeploymentConfigEnvInnerFromJSON(json) {
    return DeploymentConfigEnvInnerFromJSONTyped(json, false);
}
exports.DeploymentConfigEnvInnerFromJSON = DeploymentConfigEnvInnerFromJSON;
function DeploymentConfigEnvInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': json['value'],
        'name': json['name'],
    };
}
exports.DeploymentConfigEnvInnerFromJSONTyped = DeploymentConfigEnvInnerFromJSONTyped;
function DeploymentConfigEnvInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': value.value,
        'name': value.name,
    };
}
exports.DeploymentConfigEnvInnerToJSON = DeploymentConfigEnvInnerToJSON;
