// @hathora/hathora-cloud-sdk/dist/models/DiscoveryResponseInner.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveryResponseInnerToJSON = exports.DiscoveryResponseInnerFromJSONTyped = exports.DiscoveryResponseInnerFromJSON = exports.instanceOfDiscoveryResponseInner = void 0;
const Region_1 = require("./Region");
/**
 * Check if a given object implements the DiscoveryResponseInner interface.
 */
function instanceOfDiscoveryResponseInner(value) {
    let isInstance = true;
    isInstance = isInstance && "port" in value;
    isInstance = isInstance && "host" in value;
    isInstance = isInstance && "region" in value;
    return isInstance;
}
exports.instanceOfDiscoveryResponseInner = instanceOfDiscoveryResponseInner;
function DiscoveryResponseInnerFromJSON(json) {
    return DiscoveryResponseInnerFromJSONTyped(json, false);
}
exports.DiscoveryResponseInnerFromJSON = DiscoveryResponseInnerFromJSON;
function DiscoveryResponseInnerFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'port': json['port'],
        'host': json['host'],
        'region': (0, Region_1.RegionFromJSON)(json['region']),
    };
}
exports.DiscoveryResponseInnerFromJSONTyped = DiscoveryResponseInnerFromJSONTyped;
function DiscoveryResponseInnerToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'port': value.port,
        'host': value.host,
        'region': (0, Region_1.RegionToJSON)(value.region),
    };
}
exports.DiscoveryResponseInnerToJSON = DiscoveryResponseInnerToJSON;
