// @hathora/hathora-cloud-sdk/dist/models/LoginAnonymous200Response.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginAnonymous200ResponseToJSON = exports.LoginAnonymous200ResponseFromJSONTyped = exports.LoginAnonymous200ResponseFromJSON = exports.instanceOfLoginAnonymous200Response = void 0;
/**
 * Check if a given object implements the LoginAnonymous200Response interface.
 */
function instanceOfLoginAnonymous200Response(value) {
    let isInstance = true;
    isInstance = isInstance && "token" in value;
    return isInstance;
}
exports.instanceOfLoginAnonymous200Response = instanceOfLoginAnonymous200Response;
function LoginAnonymous200ResponseFromJSON(json) {
    return LoginAnonymous200ResponseFromJSONTyped(json, false);
}
exports.LoginAnonymous200ResponseFromJSON = LoginAnonymous200ResponseFromJSON;
function LoginAnonymous200ResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'token': json['token'],
    };
}
exports.LoginAnonymous200ResponseFromJSONTyped = LoginAnonymous200ResponseFromJSONTyped;
function LoginAnonymous200ResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'token': value.token,
    };
}
exports.LoginAnonymous200ResponseToJSON = LoginAnonymous200ResponseToJSON;
