// @hathora/hathora-cloud-sdk/dist/models/LoginGoogleRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginGoogleRequestToJSON = exports.LoginGoogleRequestFromJSONTyped = exports.LoginGoogleRequestFromJSON = exports.instanceOfLoginGoogleRequest = void 0;
/**
 * Check if a given object implements the LoginGoogleRequest interface.
 */
function instanceOfLoginGoogleRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "idToken" in value;
    return isInstance;
}
exports.instanceOfLoginGoogleRequest = instanceOfLoginGoogleRequest;
function LoginGoogleRequestFromJSON(json) {
    return LoginGoogleRequestFromJSONTyped(json, false);
}
exports.LoginGoogleRequestFromJSON = LoginGoogleRequestFromJSON;
function LoginGoogleRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'idToken': json['idToken'],
    };
}
exports.LoginGoogleRequestFromJSONTyped = LoginGoogleRequestFromJSONTyped;
function LoginGoogleRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'idToken': value.idToken,
    };
}
exports.LoginGoogleRequestToJSON = LoginGoogleRequestToJSON;
