// @hathora/hathora-cloud-sdk/dist/models/LoginNicknameRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginNicknameRequestToJSON = exports.LoginNicknameRequestFromJSONTyped = exports.LoginNicknameRequestFromJSON = exports.instanceOfLoginNicknameRequest = void 0;
/**
 * Check if a given object implements the LoginNicknameRequest interface.
 */
function instanceOfLoginNicknameRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "nickname" in value;
    return isInstance;
}
exports.instanceOfLoginNicknameRequest = instanceOfLoginNicknameRequest;
function LoginNicknameRequestFromJSON(json) {
    return LoginNicknameRequestFromJSONTyped(json, false);
}
exports.LoginNicknameRequestFromJSON = LoginNicknameRequestFromJSON;
function LoginNicknameRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'nickname': json['nickname'],
    };
}
exports.LoginNicknameRequestFromJSONTyped = LoginNicknameRequestFromJSONTyped;
function LoginNicknameRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'nickname': value.nickname,
    };
}
exports.LoginNicknameRequestToJSON = LoginNicknameRequestToJSON;
