// @hathora/hathora-cloud-sdk/dist/models/MetricValue.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricValueToJSON = exports.MetricValueFromJSONTyped = exports.MetricValueFromJSON = exports.instanceOfMetricValue = void 0;
/**
 * Check if a given object implements the MetricValue interface.
 */
function instanceOfMetricValue(value) {
    let isInstance = true;
    isInstance = isInstance && "value" in value;
    isInstance = isInstance && "timestamp" in value;
    return isInstance;
}
exports.instanceOfMetricValue = instanceOfMetricValue;
function MetricValueFromJSON(json) {
    return MetricValueFromJSONTyped(json, false);
}
exports.MetricValueFromJSON = MetricValueFromJSON;
function MetricValueFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': json['value'],
        'timestamp': json['timestamp'],
    };
}
exports.MetricValueFromJSONTyped = MetricValueFromJSONTyped;
function MetricValueToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': value.value,
        'timestamp': value.timestamp,
    };
}
exports.MetricValueToJSON = MetricValueToJSON;
