// @hathora/hathora-cloud-sdk/dist/models/MetricName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetricNameToJSON = exports.MetricNameFromJSONTyped = exports.MetricNameFromJSON = exports.MetricName = void 0;
/**
 *
 * @export
 */
exports.MetricName = {
    Cpu: 'cpu',
    Memory: 'memory',
    RateEgress: 'rate_egress',
    TotalEgress: 'total_egress'
};
function MetricNameFromJSON(json) {
    return MetricNameFromJSONTyped(json, false);
}
exports.MetricNameFromJSON = MetricNameFromJSON;
function MetricNameFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.MetricNameFromJSONTyped = MetricNameFromJSONTyped;
function MetricNameToJSON(value) {
    return value;
}
exports.MetricNameToJSON = MetricNameToJSON;
