// @hathora/hathora-cloud-sdk/dist/models/PickRoomExcludeKeyofRoomAllocations.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PickRoomExcludeKeyofRoomAllocationsToJSON = exports.PickRoomExcludeKeyofRoomAllocationsFromJSONTyped = exports.PickRoomExcludeKeyofRoomAllocationsFromJSON = exports.instanceOfPickRoomExcludeKeyofRoomAllocations = void 0;
const RoomAllocation_1 = require("./RoomAllocation");
const RoomStatus_1 = require("./RoomStatus");
/**
 * Check if a given object implements the PickRoomExcludeKeyofRoomAllocations interface.
 */
function instanceOfPickRoomExcludeKeyofRoomAllocations(value) {
    let isInstance = true;
    isInstance = isInstance && "appId" in value;
    isInstance = isInstance && "roomId" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "currentAllocation" in value;
    return isInstance;
}
exports.instanceOfPickRoomExcludeKeyofRoomAllocations = instanceOfPickRoomExcludeKeyofRoomAllocations;
function PickRoomExcludeKeyofRoomAllocationsFromJSON(json) {
    return PickRoomExcludeKeyofRoomAllocationsFromJSONTyped(json, false);
}
exports.PickRoomExcludeKeyofRoomAllocationsFromJSON = PickRoomExcludeKeyofRoomAllocationsFromJSON;
function PickRoomExcludeKeyofRoomAllocationsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'appId': json['appId'],
        'roomId': json['roomId'],
        'status': (0, RoomStatus_1.RoomStatusFromJSON)(json['status']),
        'currentAllocation': (0, RoomAllocation_1.RoomAllocationFromJSON)(json['currentAllocation']),
    };
}
exports.PickRoomExcludeKeyofRoomAllocationsFromJSONTyped = PickRoomExcludeKeyofRoomAllocationsFromJSONTyped;
function PickRoomExcludeKeyofRoomAllocationsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'appId': value.appId,
        'roomId': value.roomId,
        'status': (0, RoomStatus_1.RoomStatusToJSON)(value.status),
        'currentAllocation': (0, RoomAllocation_1.RoomAllocationToJSON)(value.currentAllocation),
    };
}
exports.PickRoomExcludeKeyofRoomAllocationsToJSON = PickRoomExcludeKeyofRoomAllocationsToJSON;
