// @hathora/hathora-cloud-sdk/dist/models/PlanName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlanNameToJSON = exports.PlanNameFromJSONTyped = exports.PlanNameFromJSON = exports.PlanName = void 0;
/**
 *
 * @export
 */
exports.PlanName = {
    Tiny: 'tiny',
    Small: 'small',
    Medium: 'medium',
    Large: 'large'
};
function PlanNameFromJSON(json) {
    return PlanNameFromJSONTyped(json, false);
}
exports.PlanNameFromJSON = PlanNameFromJSON;
function PlanNameFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.PlanNameFromJSONTyped = PlanNameFromJSONTyped;
function PlanNameToJSON(value) {
    return value;
}
exports.PlanNameToJSON = PlanNameToJSON;
