// @hathora/hathora-cloud-sdk/dist/models/ProcessWithRoomsAllOf.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessWithRoomsAllOfToJSON = exports.ProcessWithRoomsAllOfFromJSONTyped = exports.ProcessWithRoomsAllOfFromJSON = exports.instanceOfProcessWithRoomsAllOf = void 0;
const PickRoomExcludeKeyofRoomAllocations_1 = require("./PickRoomExcludeKeyofRoomAllocations");
/**
 * Check if a given object implements the ProcessWithRoomsAllOf interface.
 */
function instanceOfProcessWithRoomsAllOf(value) {
    let isInstance = true;
    isInstance = isInstance && "rooms" in value;
    isInstance = isInstance && "totalRooms" in value;
    return isInstance;
}
exports.instanceOfProcessWithRoomsAllOf = instanceOfProcessWithRoomsAllOf;
function ProcessWithRoomsAllOfFromJSON(json) {
    return ProcessWithRoomsAllOfFromJSONTyped(json, false);
}
exports.ProcessWithRoomsAllOfFromJSON = ProcessWithRoomsAllOfFromJSON;
function ProcessWithRoomsAllOfFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'rooms': (json['rooms'].map(PickRoomExcludeKeyofRoomAllocations_1.PickRoomExcludeKeyofRoomAllocationsFromJSON)),
        'totalRooms': json['totalRooms'],
    };
}
exports.ProcessWithRoomsAllOfFromJSONTyped = ProcessWithRoomsAllOfFromJSONTyped;
function ProcessWithRoomsAllOfToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'rooms': (value.rooms.map(PickRoomExcludeKeyofRoomAllocations_1.PickRoomExcludeKeyofRoomAllocationsToJSON)),
        'totalRooms': value.totalRooms,
    };
}
exports.ProcessWithRoomsAllOfToJSON = ProcessWithRoomsAllOfToJSON;
