// @hathora/hathora-cloud-sdk/dist/models/RecordPartialMetricNameMetricValueArray.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordPartialMetricNameMetricValueArrayToJSON = exports.RecordPartialMetricNameMetricValueArrayFromJSONTyped = exports.RecordPartialMetricNameMetricValueArrayFromJSON = exports.instanceOfRecordPartialMetricNameMetricValueArray = void 0;
const runtime_1 = require("../runtime");
const MetricValue_1 = require("./MetricValue");
/**
 * Check if a given object implements the RecordPartialMetricNameMetricValueArray interface.
 */
function instanceOfRecordPartialMetricNameMetricValueArray(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfRecordPartialMetricNameMetricValueArray = instanceOfRecordPartialMetricNameMetricValueArray;
function RecordPartialMetricNameMetricValueArrayFromJSON(json) {
    return RecordPartialMetricNameMetricValueArrayFromJSONTyped(json, false);
}
exports.RecordPartialMetricNameMetricValueArrayFromJSON = RecordPartialMetricNameMetricValueArrayFromJSON;
function RecordPartialMetricNameMetricValueArrayFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'cpu': !(0, runtime_1.exists)(json, 'cpu') ? undefined : (json['cpu'].map(MetricValue_1.MetricValueFromJSON)),
        'memory': !(0, runtime_1.exists)(json, 'memory') ? undefined : (json['memory'].map(MetricValue_1.MetricValueFromJSON)),
        'rateEgress': !(0, runtime_1.exists)(json, 'rate_egress') ? undefined : (json['rate_egress'].map(MetricValue_1.MetricValueFromJSON)),
        'totalEgress': !(0, runtime_1.exists)(json, 'total_egress') ? undefined : (json['total_egress'].map(MetricValue_1.MetricValueFromJSON)),
    };
}
exports.RecordPartialMetricNameMetricValueArrayFromJSONTyped = RecordPartialMetricNameMetricValueArrayFromJSONTyped;
function RecordPartialMetricNameMetricValueArrayToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'cpu': value.cpu === undefined ? undefined : (value.cpu.map(MetricValue_1.MetricValueToJSON)),
        'memory': value.memory === undefined ? undefined : (value.memory.map(MetricValue_1.MetricValueToJSON)),
        'rate_egress': value.rateEgress === undefined ? undefined : (value.rateEgress.map(MetricValue_1.MetricValueToJSON)),
        'total_egress': value.totalEgress === undefined ? undefined : (value.totalEgress.map(MetricValue_1.MetricValueToJSON)),
    };
}
exports.RecordPartialMetricNameMetricValueArrayToJSON = RecordPartialMetricNameMetricValueArrayToJSON;
