// @hathora/hathora-cloud-sdk/dist/models/Region.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegionToJSON = exports.RegionFromJSONTyped = exports.RegionFromJSON = exports.Region = void 0;
/**
 *
 * @export
 */
exports.Region = {
    Seattle: 'Seattle',
    WashingtonDc: 'Washington_DC',
    Chicago: 'Chicago',
    London: 'London',
    Frankfurt: 'Frankfurt',
    Mumbai: 'Mumbai',
    Singapore: 'Singapore',
    Tokyo: 'Tokyo',
    Sydney: 'Sydney',
    SaoPaulo: 'Sao_Paulo'
};
function RegionFromJSON(json) {
    return RegionFromJSONTyped(json, false);
}
exports.RegionFromJSON = RegionFromJSON;
function RegionFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.RegionFromJSONTyped = RegionFromJSONTyped;
function RegionToJSON(value) {
    return value;
}
exports.RegionToJSON = RegionToJSON;
