// @hathora/hathora-cloud-sdk/dist/models/RoomStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoomStatusToJSON = exports.RoomStatusFromJSONTyped = exports.RoomStatusFromJSON = exports.RoomStatus = void 0;
/**
 *
 * @export
 */
exports.RoomStatus = {
    Scheduling: 'scheduling',
    Active: 'active',
    Suspended: 'suspended',
    Destroyed: 'destroyed'
};
function RoomStatusFromJSON(json) {
    return RoomStatusFromJSONTyped(json, false);
}
exports.RoomStatusFromJSON = RoomStatusFromJSON;
function RoomStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.RoomStatusFromJSONTyped = RoomStatusFromJSONTyped;
function RoomStatusToJSON(value) {
    return value;
}
exports.RoomStatusToJSON = RoomStatusToJSON;
