// @hathora/hathora-cloud-sdk/dist/models/Room.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoomToJSON = exports.RoomFromJSONTyped = exports.RoomFromJSON = exports.instanceOfRoom = void 0;
const RoomAllocation_1 = require("./RoomAllocation");
const RoomStatus_1 = require("./RoomStatus");
/**
 * Check if a given object implements the Room interface.
 */
function instanceOfRoom(value) {
    let isInstance = true;
    isInstance = isInstance && "currentAllocation" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "allocations" in value;
    isInstance = isInstance && "roomId" in value;
    isInstance = isInstance && "appId" in value;
    return isInstance;
}
exports.instanceOfRoom = instanceOfRoom;
function RoomFromJSON(json) {
    return RoomFromJSONTyped(json, false);
}
exports.RoomFromJSON = RoomFromJSON;
function RoomFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'currentAllocation': (0, RoomAllocation_1.RoomAllocationFromJSON)(json['currentAllocation']),
        'status': (0, RoomStatus_1.RoomStatusFromJSON)(json['status']),
        'allocations': (json['allocations'].map(RoomAllocation_1.RoomAllocationFromJSON)),
        'roomId': json['roomId'],
        'appId': json['appId'],
    };
}
exports.RoomFromJSONTyped = RoomFromJSONTyped;
function RoomToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'currentAllocation': (0, RoomAllocation_1.RoomAllocationToJSON)(value.currentAllocation),
        'status': (0, RoomStatus_1.RoomStatusToJSON)(value.status),
        'allocations': (value.allocations.map(RoomAllocation_1.RoomAllocationToJSON)),
        'roomId': value.roomId,
        'appId': value.appId,
    };
}
exports.RoomToJSON = RoomToJSON;
