// @hathora/hathora-cloud-sdk/dist/models/SetLobbyStateRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SetLobbyStateRequestToJSON = exports.SetLobbyStateRequestFromJSONTyped = exports.SetLobbyStateRequestFromJSON = exports.instanceOfSetLobbyStateRequest = void 0;
/**
 * Check if a given object implements the SetLobbyStateRequest interface.
 */
function instanceOfSetLobbyStateRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "state" in value;
    return isInstance;
}
exports.instanceOfSetLobbyStateRequest = instanceOfSetLobbyStateRequest;
function SetLobbyStateRequestFromJSON(json) {
    return SetLobbyStateRequestFromJSONTyped(json, false);
}
exports.SetLobbyStateRequestFromJSON = SetLobbyStateRequestFromJSON;
function SetLobbyStateRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'state': json['state'],
    };
}
exports.SetLobbyStateRequestFromJSONTyped = SetLobbyStateRequestFromJSONTyped;
function SetLobbyStateRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'state': value.state,
    };
}
exports.SetLobbyStateRequestToJSON = SetLobbyStateRequestToJSON;
