// @hathora/hathora-cloud-sdk/dist/models/StartingConnectionInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StartingConnectionInfoToJSON = exports.StartingConnectionInfoFromJSONTyped = exports.StartingConnectionInfoFromJSON = exports.instanceOfStartingConnectionInfo = exports.StartingConnectionInfoStatusEnum = void 0;
/**
 * @export
 */
exports.StartingConnectionInfoStatusEnum = {
    Starting: 'starting'
};
/**
 * Check if a given object implements the StartingConnectionInfo interface.
 */
function instanceOfStartingConnectionInfo(value) {
    let isInstance = true;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "roomId" in value;
    return isInstance;
}
exports.instanceOfStartingConnectionInfo = instanceOfStartingConnectionInfo;
function StartingConnectionInfoFromJSON(json) {
    return StartingConnectionInfoFromJSONTyped(json, false);
}
exports.StartingConnectionInfoFromJSON = StartingConnectionInfoFromJSON;
function StartingConnectionInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'status': json['status'],
        'roomId': json['roomId'],
    };
}
exports.StartingConnectionInfoFromJSONTyped = StartingConnectionInfoFromJSONTyped;
function StartingConnectionInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'status': value.status,
        'roomId': value.roomId,
    };
}
exports.StartingConnectionInfoToJSON = StartingConnectionInfoToJSON;
