// @hathora/hathora-cloud-sdk/dist/models/TransportType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Hathora Cloud API
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransportTypeToJSON = exports.TransportTypeFromJSONTyped = exports.TransportTypeFromJSON = exports.TransportType = void 0;
/**
 *
 * @export
 */
exports.TransportType = {
    Tcp: 'tcp',
    Udp: 'udp',
    Tls: 'tls'
};
function TransportTypeFromJSON(json) {
    return TransportTypeFromJSONTyped(json, false);
}
exports.TransportTypeFromJSON = TransportTypeFromJSON;
function TransportTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.TransportTypeFromJSONTyped = TransportTypeFromJSONTyped;
function TransportTypeToJSON(value) {
    return value;
}
exports.TransportTypeToJSON = TransportTypeToJSON;
